extensions/smallrye-fault-tolerance/runtime/src/main/java/io/quarkus/smallrye/faulttolerance/runtime/SmallryeFaultToleranceTemplate.java

```diff
@@ -0,0 +1,30 @@
+package io.quarkus.smallrye.faulttolerance.runtime;
+
+import com.netflix.hystrix.Hystrix;
+import com.netflix.hystrix.metric.HystrixCollapserEventStream;
+import com.netflix.hystrix.metric.HystrixCommandCompletionStream;
+import com.netflix.hystrix.metric.HystrixCommandStartStream;
+import com.netflix.hystrix.metric.HystrixThreadEventStream;
+import com.netflix.hystrix.metric.HystrixThreadPoolCompletionStream;
+
+import io.quarkus.runtime.ShutdownContext;
+import io.quarkus.runtime.annotations.Template;
+
+@Template
+public class SmallryeFaultToleranceTemplate {
+
+    public void resetCommandContextOnUndeploy(ShutdownContext context) {
+        context.addShutdownTask(new Runnable() {
+            @Override
+            public void run() {
+                HystrixCommandCompletionStream.reset();
+                HystrixCollapserEventStream.reset();
+                HystrixCommandStartStream.reset();
+                HystrixThreadPoolCompletionStream.reset();
+                HystrixCommandStartStream.reset();
+                HystrixThreadEventStream.getInstance().shutdown();
+                Hystrix.reset();
```

Comment:
`Hystrix.reset()` should not be necessary as we call it when the CDI app context is destroyed - https://github.com/smallrye/smallrye-fault-tolerance/blob/master/implementation/src/main/java/io/smallrye/faulttolerance/HystrixInitializer.java#L64. I'm not sure about the streams though. But better safe than sorry.